### ISSUE TITLE
3.1.2 breaks hot reload for my setup

### ISSUE THREAD

**Issue by bes** [webpack/webpack-dev-server#1366]
* Operating System: macOS 
* Node Version: 9.11.1
* NPM Version: 5.6.0
* webpack Version: 4.5.0
* webpack-dev-server Version: 3.1.2

- [x] This is a **bug**
- [ ] This is a **modification** request

webpack-dev-server v3.1.2 breaks hot reloading for my app with the following error in the browser:

```
only-dev-server.js:9 
Uncaught TypeError: Cannot read property 'indexOf' of undefined
    at upToDate (only-dev-server.js:9)
    at EventEmitter.<anonymous> (only-dev-server.js:87)
    at EventEmitter.push../node_modules/events/events.js.EventEmitter.emit (events.js:81)
    at reloadApp (client?fd53:226)
    at Object.ok (client?fd53:139)
    at SockJS.onmessage (socket.js:41)
    at SockJS.EventTarget.dispatchEvent (eventtarget.js:51)
    at main.js:274
    at Array.forEach (<anonymous>)
    at SockJS._transportMessage (main.js:272)
```

The code at only-dev-server.js:9 looks like this

```
if (module.hot) {
	var lastHash;
	var upToDate = function upToDate() {
		return lastHash.indexOf(__webpack_hash__) >= 0;
	};
```

the crash at lastHash being undefined when calling indexOf would lead me to believe that 

```
    hotEmitter.on("webpackHotUpdate", function(currentHash) {
```

was never called.

Switching back to 3.1.1 fixes the problem.

### Code

My setup uses splitChunks enabled, and the error message would lead me to believe that the hash for the updated chunk could not be found?

```
        splitChunks: {
          name: true,
          chunks: "all",
        },
```

A minimal proof-of-concept can be found here:
https://github.com/bes/webpack-dev-server-3.1.2-bug-hot-reload

### Expected Behavior
Hot reload to work

### Actual Behavior
Hot reload crashes with the above stack trace.

### For Bugs; How can we reproduce the behavior?
Use the POC linked above

**Comment by bes** [comment]
The only patches between 3.1.1 and 3.1.2 are
* https://github.com/webpack/webpack-dev-server/pull/1362
* https://github.com/webpack/webpack-dev-server/commit/50f40077d00fbbda449407b97e06655934264c42

**Comment by b12k** [comment]
+1
Facing sama problem, with the same setup but on Windows 10.

**Comment by SolidZORO** [comment]
+1.

"webpack-dev-server": "^3.1.1"

**Comment by tusharc88** [comment]
+1

**Comment by happystory** [comment]
I encountered the same problem, how to fixed it?

**Comment by SolidZORO** [comment]
@happystory  I just rollback "webpack-dev-server": "3.1.0", is work and waitting bug fixed.

**Comment by Topppy** [comment]
+1
"webpack-dev-server": "^3.1.2"

**Comment by aap82** [comment]
+1 on 'lastHash is undefined' with 3.1.2

**Comment by SpaceK33z** [comment]
Thanks for your clear report. It should be fixed with `webpack-dev-server@3.1.3`.
